app/joinCircle.js

```diff
@@ -50,78 +54,78 @@ export default function Page() {
 	}
 
 	return (
-		<StyledSafeArea className='bg-offblack  flex-1'>
+		<StyledSafeArea
+			className='bg-offblack flex-1 h-screen'
+			style={{ paddingTop: topInset }}
+		>
 			<KeyboardAwareScrollView bounces={false}>
-				<StyledView className='flex items-center justify-center text-center w-screen h-[90px]'>
-					<StyledText className='text-offwhite font-bold text-4xl'>
-						Join New Circle:
-					</StyledText>
-				</StyledView>
-				<StyledView className='flex-row justify-center'>
-					<StyledView className='w-[300px] h-[300px] border-2 border-offwhite'>
-						<BarCodeScanner
-							mirrorImage={true}
-							fixOrientation={true}
-							className='w-full h-full'
-							ratio='1:1'
-							onBarCodeScanned={
-								scanned ? undefined : handleBarCodeScanned
-							}
-						/>
+				<StyledView className='flex-1 items-center h-screen'>
+					<StyledView className='flex items-center justify-center text-center w-screen h-[90px]'>
+						<StyledText className='text-offwhite font-bold text-4xl'>
+							Join New Circle
+						</StyledText>
 					</StyledView>
+
+					<StyledView className='absolute top-[15%] border-[10px] border-offwhite rounded-xl'>
+						<StyledView className='w-[300px] h-[300px]'>
+							<BarCodeScanner
+								mirrorImage={true}
+								fixOrientation={true}
+								className='w-full h-full'
+								ratio='1:1'
+								onBarCodeScanned={
+									scanned ? undefined : handleBarCodeScanned
+								}
+							/>
+						</StyledView>
+					</StyledView>
+					<StyledView className='absolute w-[70%] bottom-[30%] border-[10px] border-offwhite mt-20 rounded-xl'>
+						<StyledInput
+							className='bg-offblack px-[10px] font-bold text-2xl flex-1 h-[42px] text-offwhite rounded-lg py-[5px] mr-1'
+							placeholder={'Code:'}
+							placeholderTextColor={'#FFFBFC'}
+							inputMode='numeric'
+							maxLength={8}
+							onChangeText={(text) => {
+								setCode(text);
+							}}
+							ref={(input) => {
+								this.searchCode = input;
+							}}
+						/>
+					</StyledView>				
 				</StyledView>
-				<StyledView className='flex items-center justify-center text-center w-screen h-[90px]'>
-					<StyledText className='text-offwhite font-bold text-4xl'>
-						Circle Code:
-					</StyledText>
-				</StyledView>
-				<StyledView className='flex items-center justify-center'>
-					<StyledInput
-						className='bg-offblack text-[18px] flex-1 w-[300px] h-[42px] text-offwhite border border-offwhite rounded-lg px-3 py-[5px] mr-1'
-						placeholder={'Code:'}
-						placeholderTextColor={'#ffffff66'}
-						inputMode='numeric'
-						maxLength={8}
-						onChangeText={(text) => {
-							setCode(text);
+				<StyledView
```

Comment:
This view needs to be outside of the scroll view for the buttons to be visible all the time, following the structure of all other pages